src/PollinationSDK/Client/Configuration.cs

```diff
@@ -14,9 +14,69 @@
 using System.IO;
 using System.Linq;
 using System.Text;
+using RestSharp;
 
 namespace PollinationSDK.Client
 {
+    /// <summary>
+    /// TokenRepo is responsible for holding and acquiring auth tokens
+    /// </summary>
+    public class TokenRepo
+    {
+        public string RefreshURL;
+        private string IDToken;
+        public DateTime ExpiresAt;
+        public string RefreshToken;
+
+        public TokenRepo(string refreshURL, string idToken, int expiresInSeconds, string refreshToken)
+        {
+            RefreshURL = refreshURL;
+            IDToken = idToken;
+            ExpiresAt = DateTime.Now.AddSeconds(expiresInSeconds);
+            RefreshToken = refreshToken;
+        }
+
+        private async System.Threading.Tasks.Task RefreshTokenAsync()
+        {
+
+            var client = new RestClient(this.RefreshURL);
+
+            var req = new RestRequest().AddJsonBody(
+                new Dictionary<string, string>
+                {
+                    {
+                        "token", this.RefreshToken }
+                    }
+                );
+
+            var res = await client.PostAsync<Dictionary<string, string>>(req);
+
+            string detail = null;
+
+            res.TryGetValue("detail", out detail);
+            res.TryGetValue("message", out detail);
```

Comment:
Is this correct? You are assigning values to the same variable.

Reply:
Yeah, the response fields are exclusive. In case of an error, the pollination APIs use `detail` and the Firebase responses use `message` to describe what happened. I was trying to keep the API simple by just passing the responses through unchanged, but I can abstract them both to `detail` if you think it's better.

Reply:
Added a comment in the template to clarify.